android/libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt

```diff
@@ -348,4 +360,79 @@ open class Node<V : RibView>(
 
     override fun toString(): String =
         "Node@${hashCode()} ($identifier)"
+
+    /**
+     * Executes an action and remains on the same hierarchical level
+     *
+     * @return the current workflow element
+     */
+    protected inline fun <reified T> executeWorkflow(
+        crossinline action: () -> Unit
+    ): Single<T> = Single.fromCallable {
+            action()
+            this as T
+        }
+        .takeUntil(detachSignal.firstOrError())
+
+    @VisibleForTesting
```

Comment:
can we access these methods by extending `Node` class in tests and exposing protected methods?

Reply:
Unfortunately not, it gives me compiler error, because they are inlined. Unless you know another way.